content/docs/ref/pr.md

```diff
@@ -2,18 +2,29 @@
 
 Commit specified files to a new branch and create a pull request.
 
-```dvc
-cml pr '**/*.py' '**/*.json'
+```bash
+cml pr "**/*.py" "**/*.json"
 ```
 
 is roughly equivalent to:
 
-```dvc
+```bash
 SHA="$(git log -n1 --format=%h)"
 BASE="$(git branch)"
 git checkout -b "${BASE}-cml-pr-${SHA}"
-git add *.py *.json
+find . -name '*.py' -o -name '*.json' | xargs git add
 git commit -m "CML PR for ${SHA} [skip ci]"
 git push
-gh pr create -B ${BASE}
+curl \
+  -X POST \
+  -H "Accept: application/vnd.github.v3+json" \
+  https://api.github.com/repos/${GITHUB_REPOSITORY}/pulls \
+  -d "{
+    \"head\": \"${BASE}-cml-pr-${SHA}\",
+    \"base\": \"${BASE}\",
+    \"title\": \"CML PR for ${BASE} ${SHA}\",
+    \"description\":
+      \"Automated commits for ${GITHUB_REPOSITORY}/commit/${SHA} created by CML.\"
+  }" \
+  | jq -r .url
```

Comment:
CC @DavidGOrtega happy now? :D

Reply:
🤯 
I still say is not the same. If you repeat the workflow you will have an error
😎

Reply:
Basically we retrieve the PR first and if do not exists we do all the push, etc...

Reply:
As per [`gitglossary(7)`](https://man7.org/linux/man-pages/man7/gitglossary.7.html) path specification documentation, it should be possible to `git add "**/*.py" "**/*.json"` without resorting to extended globs or `find | xargs` alternatives. 🤔

Reply:
actually no :scream: if the branch already exists we do nothing!

https://github.com/iterative/cml/blob/9909e2a809af0467f5b2f3cb0f08ff51c07d0cbb/src/cml.js#L297-L305

Reply:
Are you pointing out that for cml to not have to use globs itself  and instead make use of Git's capabilities?

Reply:
> actually no 😱 if the branch already exists we do nothing!

and thats correct

Reply:
> _Are you pointing out that for cml to not have to use globs itself and instead make use of Git's capabilities?_

Not necessarily, especially if we go full-JavaScript for Git. My comment was just about the alternative “explanation” of the command behavior, not about the internal `cml pr` implementation.

Reply:
> > actually no scream if the branch already exists we do nothing!
> 
> and thats correct, isnt it?

super un-intuitive, but at the very least needs documenting. Better now?

Reply:
> super un-intuitive, but at the very least needs documenting. Better now?

Not really! If you try to create a PR attached to a commit the SCM will close previous one. Which is in my opinion even less intuitive. Aside that the add, commit etc will fail because the branch already exist.
I found that cml pr returns you a the PR associated to that commit or it creates it.

Reply:
multiple problems (but not for this docs PR):

- if `"${BASE}-cml-pr-${SHA}"` was manually created (exists but there is no associated PR) then `cmp pr` royally fails
- `cml pr` should complain (exit non-zero) if a PR already exists, and also have a `--force` flag

Reply:
> if "${BASE}-cml-pr-${SHA}" was manually created

Is this scenario plausible? maybe then we should use a less plausible naming convention? 

> cml pr should complain (exit non-zero) if a PR already exists, and also have a --force flag

not very CI friendly and not sure if this is helpful in any case in my opinion. I do not see an useful scenario for this.

Reply:
opened [cml#758](https://github.com/iterative/cml/issues/758) to track :)